### PR TITLE
Update data for WebOTP API

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

This PR adds some compat data that was missing for the [WebOTP API](https://wicg.github.io/web-otp/). You can find more information about this API at [Verify phone numbers on the web with the WebOTP API](https://developer.chrome.com/articles/web-otp/),

Also see my [research document](https://docs.google.com/document/d/1vPnLk_CF4LNc0AVl5ieqRv5TKqCroJ8hPCPp9rfTaxw/edit#) for details of exactly what changes have been made in this PR.

### CODE REVIEW

File: html/global_attributes.json [hunk 1]
```diff
@@ -150,6 +150,43 @@
             }
           }
         },
+        "one-time-code": {
+          "__compat": {
+            "description": "<code>new-password</code> value",
```
Comment (Elchi3) [inline html/global_attributes.json:155]:
```suggestion
            "description": "<code>one-time-code</code> value",
```

File: html/global_attributes.json [hunk 2]
```diff
@@ -150,6 +150,43 @@
...
+              },
+              "firefox_android": "mirror",
+              "ie": {
+                "version_added": false
+              },
+              "oculus": "mirror",
+              "opera": "mirror",
+              "opera_android": "mirror",
+              "safari": {
+                "version_added": "14"
```
Comment (Elchi3) [inline html/global_attributes.json:175]: This is in Safari? Source?

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks! A suggestion and a question for you.

File: html/global_attributes.json (hunk 2 above)
Comment (chrisdavidmills) [reply html/global_attributes.json:175]: See the second paragraph below the heading at https://developer.chrome.com/articles/web-otp/#annotate-an-input-tag

Comment (Elchi3) [reply html/global_attributes.json:175]: Actually, it should be 12 per https://github.com/whatwg/html/issues/3745#issuecomment-395147558


```suggestion
                "version_added": "12"
```

**Review by Elchi3 — APPROVED** [review]
Nice! Thanks @chrisdavidmills 👍

**Pull request merged**